specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-10-01-preview/containerInstance.json

```diff
@@ -0,0 +1,977 @@
+          "type": "string",
+          "description": "The protocol associated with the port.",
+          "enum": [
+            "TCP",
+            "UDP"
+          ],
+          "x-ms-enum": {
+            "name": "ContainerNetworkProtocol",
+            "modelAsString": true
+          }
+        },
+        "port": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The port number exposed within the container group."
+        }
+      },
+      "required": [
+        "port"
+      ]
+    },
+    "EnvironmentVariable": {
+      "description": "The environment variable to set within the container instance.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the environment variable."
+        },
+        "value": {
+          "type": "string",
+          "description": "The value of the environment variable."
+        }
+      },
+      "required": [
+        "name",
+        "value"
+      ]
+    },
+    "OperationListResult":{
+      "description": "The operation list response that contains all operations for Azure Container Instance service.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Operation"
+          },
+          "description": "The list of operations."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URI to fetch the next page of operations."
+        }
+      }
+    },
+    "Operation": {
+      "description": "An operation for Azure Container Instance service.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the operation."
+        },
+        "display": {
+          "description": "The display information of the operation.",
+          "type": "object",
+          "properties": {
+            "provider": {
+              "description": "The name of the provider of the operation.",
+              "type": "string"
+            },
+            "resource": {
+              "type": "string",
+              "description": "The name of the resource type of the operation."
+            },
+            "operation": {
+              "description": "The friendly name of the operation.",
+              "type": "string"
+            },
+            "description": {
+              "description": "The description of the operation.",
+              "type": "string"
+            }
+          }
+        },
+        "origin": {
+          "type": "string",
+          "description": "The intended executor of the operation.",
+          "enum": [
+            "User",
+            "System"
+          ],
+          "x-ms-enum": {
+            "name": "ContainerInstanceOperationsOrigin",
+            "modelAsString": true
+          }
+        }
+      },
+      "required": [
```

Comment:
>"required": [ [](start = 6, length = 13)

Question, is this required server side?

Reply:
From the code, I don't see them required. Should I remove?

Reply:
What I mean here is, these two are marked as required, but from this swagger I see this model is only used when it is returned by the service, not sent by the client. That's why I was wondering why they required. Is there maybe an `Operation_Create` operation missing in the swagger?

Reply:
I see. So "required" is only for user input. Will remove.